### PR TITLE
Feature/31 allow users to sign up

### PR DESCRIPTION
**Pull request by ostartt**

### CODE REVIEW

File: src/main/java/com/softserve/edu/sporthubujp/controller/ArticleController.java [hunk 1]
```diff
@@ -24,7 +21,7 @@ public ArticleController(ArticleService articleService) {
     }
 
     @GetMapping("/{id}")
-    @PreAuthorize("hasAnyAuthority('ADMIN', 'USER')")
+    @PreAuthorize("hasAuthority('USER')")
```
Comment (yuremboo) [inline src/main/java/com/softserve/edu/sporthubujp/controller/ArticleController.java:24]: article can read anybody

File: src/main/java/com/softserve/edu/sporthubujp/entity/ConfirmationToken.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+            name = "confirmation_token_sequence",
+            sequenceName = "confirmation_token_sequence",
+            allocationSize = 1
+    )
+    @Id
+    @GeneratedValue(
+            strategy = GenerationType.SEQUENCE,
+            generator = "confirmation_token_sequence"
+    )
+    private Long id;
```
Comment (yuremboo) [inline src/main/java/com/softserve/edu/sporthubujp/entity/ConfirmationToken.java:27]: keep consistency with other tables, use UUID

File: src/main/java/com/softserve/edu/sporthubujp/entity/ConfirmationToken.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+    @Column(nullable = false)
+    private LocalDateTime createdAt;
+
+    @Column(nullable = false)
+    private LocalDateTime expiresAt;
+
+    private LocalDateTime confirmedAt;
+
+    @ManyToOne
```
Comment (yuremboo) [inline src/main/java/com/softserve/edu/sporthubujp/entity/ConfirmationToken.java:40]: @OneToOne ?

File: src/main/java/com/softserve/edu/sporthubujp/repository/ConfirmationTokenRepository.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+import org.springframework.data.jpa.repository.Modifying;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.time.LocalDateTime;
+import java.util.Optional;
+
+@Repository
+@Transactional(readOnly = true)
```
Comment (yuremboo) [inline src/main/java/com/softserve/edu/sporthubujp/repository/ConfirmationTokenRepository.java:14]: @ Transactional required only on services

File: src/main/java/com/softserve/edu/sporthubujp/repository/ConfirmationTokenRepository.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+public interface ConfirmationTokenRepository
+        extends JpaRepository<ConfirmationToken, Long> {
+
+    Optional<ConfirmationToken> findByToken(String token);
+
+    @Transactional
```
Comment (yuremboo) [inline src/main/java/com/softserve/edu/sporthubujp/repository/ConfirmationTokenRepository.java:20]: @ Transactional required only on services

File: src/main/java/com/softserve/edu/sporthubujp/repository/UserRepository.java [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Modifying;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.Optional;
+
+@Repository
+@Transactional(readOnly = true)
```
Comment (yuremboo) [inline src/main/java/com/softserve/edu/sporthubujp/repository/UserRepository.java:13]: Transactional required only on services

File: src/main/java/com/softserve/edu/sporthubujp/service/impl/EmailServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+import org.springframework.stereotype.Service;
+
+import javax.mail.MessagingException;
+import javax.mail.internet.MimeMessage;
+
+@Service
+@AllArgsConstructor
+public class EmailServiceImpl implements EmailSenderService {
+
+    private final static Logger LOGGER = LoggerFactory
```
Comment (yuremboo) [inline src/main/java/com/softserve/edu/sporthubujp/service/impl/EmailServiceImpl.java:19]: you can use Lombok @ Slf4j

File: src/main/java/com/softserve/edu/sporthubujp/service/impl/RegistrationService.java [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+        }
+
+        confirmationTokenService.setConfirmedAt(token);
+        userService.enableUser(
+                confirmationToken.getUser().getEmail());
+        return "confirmed";
+    }
+
+    private String buildEmail(String name, String link) {
+        return "<div style=\"font-family:Helvetica,Arial,sans-serif;font-size:16px;margin:0;color:#0b0c0c\">\n" +
```
Comment (yuremboo) [inline src/main/java/com/softserve/edu/sporthubujp/service/impl/RegistrationService.java:74]: Use .html file in resources for this

File: src/main/java/com/softserve/edu/sporthubujp/service/impl/UserServiceImpl.java [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+import com.softserve.edu.sporthubujp.service.UserService;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDateTime;
+import java.util.UUID;
+
+@Service
+@AllArgsConstructor
+public class UserServiceImpl implements UserService {
```
Comment (yuremboo) [inline src/main/java/com/softserve/edu/sporthubujp/service/impl/UserServiceImpl.java:18]: Add logs

**Review by yuremboo — CHANGES_REQUESTED** [review]

**Review by yuremboo — APPROVED** [review]

**Review by AndriyBarskyi — APPROVED** [review]

**Pull request merged**